Marlin/src/HAL/STM32/u8g/LCD_defines.h

```diff
@@ -25,8 +25,13 @@
  * STM32 LCD-specific defines
  */
 
-uint8_t u8g_com_std_sw_spi_fn(u8g_t *u8g, uint8_t msg, uint8_t arg_val, void *arg_ptr);         // See U8glib-HAL
-uint8_t u8g_com_stm32duino_hw_spi_fn(u8g_t *u8g, uint8_t msg, uint8_t arg_val, void *arg_ptr);  // See U8glib-HAL
+#if ALL(HAS_MARLINUI_U8GLIB, FORCE_SOFT_SPI)
```

Comment:
Does this work as intended?

One of the problematic hardware configurations I have (and was hoping this PR would help) is on a BIQU Hurakan with a `BTT_MINI_12864` & `BOARD_BTT_MANTA_M4P_V2_1` (STM32G0).

Both `HAS_MARLINUI_U8GLIB` & `FORCE_SOFT_SPI` are set elsewhere, but both are true...and not... ahead of this (`LCD_defines.h`) file due to file include order(?).

I threw in some errors as a check:

<img width="949" alt="image" src="https://github.com/MarlinFirmware/Marlin/assets/13375512/c9e46202-32e7-4430-8ca7-54438c4dd53b">

Both of my errors show up in the terminal.

In either case, it doesn't fix the SPI issues I'm having with this LCD/hardware combination (bummer!), but I thought I'd mention it.

Reply:
Also just to double check: you should be getting a screen with random dots at least on powerup if everything is wired correctly.  On my BTT display the plugs were upside down and the arrow that normally points at pin 1 points at pin 10 instead.  I had to wiggle the plugs off and put them back on backwards before it'd work.  The FYSEC display I have had them facing the right direction out of the box.

Reply:
> Does that look any better for that board and display combo now?

There's no change in the M4P/Mini 12864 display behavior with this PR, so it's probably something else.

> Also just to double check: you should be getting a screen with random dots at least on powerup if everything is wired correctly. 

Yep, no issues there since the Manta M4P has two EXP connectors.

> On my BTT display the plugs were upside down and the arrow that normally points at pin 1 points at pin 10 instead. I had to wiggle the plugs off and put them back on backwards before it'd work. The FYSEC display I have had them facing the right direction out of the box.

No one can agree on what is Pin 1, so it's common to rotate EXP cables 180° when using mixed & matched hardware. Most of my EXP cables have the tabs cut off because of it 😄

Reply:
Just wondering: does the SD card slot work on your display?

Reply:
> does the SD card slot work on your display?

Sort of.

The screen doesn't update and requires the encoder to be pressed to "refresh" the screen. This initially happens after displaying the Marlin & custom boot logos and status screen doesn't show / appears frozen, but clicking the encoder refreshes the screen. Inserting an SD card into the LCD's SD card reader shows "Media Inserted" on the status screen, but the screen will lock up if I try to access the SD card until I remove it. Once the SD card is removed, the screen "works" again.

Setting `SDCARD_CONNECTION` to `ONBOARD` doesn't work at all. In fact, the motherboard reboots when you insert an SD card to the onboard SD slot.

Note: The Manta series of boards were designed with Klipper in mind originally, but Marlin support was ported to them, so some quirks are not unexpected. Speaking of Klipper, inserting an SD card into the onboard SD slot does not cause the motherboard to reboot, but Klipper doesn't use the onboard SD slot for anything, so 🤷

Reply:
Behavior is the same, so I'll stop hijacking your PR 😄

I'll open a proper bug report for this at some point, but not many people are running Marlin on this hardware combination for it to be an issue apparently.